content/advent-2018/modules-and-athens.md

```diff
@@ -0,0 +1,214 @@
+to release a new version of our code. This architecture allows us to keep 
+using those tools and that workflow. 
+
+>Note: if you're not using [semver](https://semver.org) tags to release new 
+versions of your module, you should start doing that now!
+
+And by building _proxy servers_, we don't have to change anything about our 
+workflow to fetch modules. A simple `go get` against a proxy server will 
+fetch a module at the version of your choice.
+
+## Introducing Athens
+
+Now that there's a high level architecture for module proxies, the question 
+is what should a proxy look like? There are two main possibilities:
+
+- A CDN backed by static storage like a cloud blob store or a filesystem
+- A server, backed by a database, that fetches modules from an upstream VCS 
+host as needed
+
+In order to not disturb workflows like discussed in the previous section, we 
+need to implement the second one, and Athens is the first open source 
+implementation to do that.
+
+In other words, Athens is an immutable mirror of upstream VCS hosts. When you 
+execute `go get github.com/athens-artifacts/maturelib@v1.0.0` against an 
+Athens proxy, Athens will store [this code](https://github.com/athens-artifacts/maturelib/releases/tag/v0.0.1) in its 
+database forever, regardless of what happens to that code on GitHub.
+
+![athens-diagram](/postimages/advent-2018/go-modules-athens/athens-diagram.png)
+
+## Experiences with Athens
+
+We've tested Athens out in lots of different situations:
+
+- As a server on `localhost`
+- A server just for a team's private code
+- A public, global proxy (experimental)
+
+As we've gathered feedback and reviewed our experiences in each situation, 
+we've found some common benefits.
+
+### Builds are Faster
+
+Go modules maintains a local on-disk cache (at `$GOPATH/pkg/mod`), which of 
+course makes builds faster than fetching modules from the network! But, for 
+cold caches (like in CI/CD systems), builds are faster than with 
+fetch-from-VCS situations.
+
+We get speed for a pretty simple reason - Athens serves zip files with module 
+code in them, and no VCS history. It's the same thing as if you ran 
+`git clone --depth 1` and then zipped up the result.
+
+### A Vendor in The Cloud
+
+Athens holds an immutable database, so we're insulated from changes to 
+upstream repositories, even if we delete the `vendor` directory. Simply 
+put, that lets us delete code from our repository while still keeping our 
+builds fast and, more importantly, reliable.
+
+### HTTP Download API
+
+Because the download API is HTTP, we can take advantage of all the battle 
+tested web technologies we know and love. For example, internal teams can 
+load balance across many Athens servers, and we can put a global caching 
+proxy (a proxy in a proxy - inception!) in front of the public proxy.
+
+### Federation
+
+Federation is important tome because it promotes collaboration and open-ness. 
```

Comment:
minor type
```suggestion
Federation is important to me because it promotes collaboration and open-ness. 
```

Reply:
Thanks @jcaberio - I hadn't tried out the suggested change feature yet either, so double thanks 😄